src/LorentzVectorHEP.jl

```diff
@@ -4,103 +4,26 @@ using LinearAlgebra
 
 export LorentzVectorCyl, LorentzVector
 
-export px, py, pz, energy, fast_mass, pt, rapidity, eta, phi, mass
-export deltaphi, deltar, deltaeta
-export ΔR, Δϕ, Δη
-export fromPtEtaPhiE
-
-
-"""
-    LorentzVector(t, x, y, z)
-
-Lorentz 4-vector, as used in Special Relativity.
-
-The metric convention is g = diag(+1,-1,-1,-1). No distinction is made between
-co- and contravariant vectors.
-"""
-struct LorentzVector{T <: AbstractFloat}
-    t :: T
-    x :: T
-    y :: T
-    z :: T
-end
-
-"""
-    LorentzVector(t, x, y, z)
-
-Promoting constructors for LorentzVector{T}.
-"""
-LorentzVector(t, x, y, z) = LorentzVector(promote(t, x, y, z)...)
-LorentzVector(t::T, x::T, y::T, z::T) where {T <: Union{Integer, Rational, Irrational}} =
-    LorentzVector(float(t), x, y, z)
-
-"""
-    dot(u, v)
-    u⋅v
-
-Inner product of 4-vectors, in the Minkowsky metric (+,-,-,-).
-"""
-function LinearAlgebra.dot(u::LorentzVector, v::LorentzVector)
-    @fastmath u.t*v.t - u.x*v.x - u.y*v.y - u.z*v.z
-end
-
-
-function Base.:(+)(u::LorentzVector, v::LorentzVector)
-    @fastmath LorentzVector(u.t + v.t, u.x + v.x, u.y + v.y, u.z + v.z)
-end
-
-function Base.:(-)(u::LorentzVector)
-    @fastmath LorentzVector(-u.t, -u.x, -u.y, -u.z)
-end
-
-function Base.:(-)(u::LorentzVector, v::LorentzVector)
-    @fastmath u + (-v)
-end
-
-function Base.:(*)(λ::Number, u::LorentzVector)
-    @fastmath LorentzVector(λ*u.t, λ*u.x, λ*u.y, λ*u.z)
-end
-
-function Base.:(*)(u::LorentzVector, λ::Number)
-    @fastmath λ * u
-end
-
-function Base.:(/)(u::LorentzVector, λ::Number)
-    @fastmath u * (one(λ) / λ)
-end
+import LorentzVectorBase: px,py,pz,energy,pt,eta,phi,mass,rapidity
+export px,py,pz,energy,pt,eta,phi,mass,rapidity
 
-function Base.:(==)(u::LorentzVector, v::LorentzVector)
-    u.t == v.t && u.x == v.x && u.y == v.y && u.z == v.z
-end
+export deltaphi, deltar, deltaeta
 
-function Base.isapprox(u::LorentzVector, v::LorentzVector;
-                  atol::Real=0,
-                  rtol::Real=atol>0 ? 0 : √min(eps(typeof(u.x)), eps(typeof(v.x))))
+export fast_mass
 
-    err = max(abs(u.t - v.t), sqrt((u.x - v.x)^2 + (u.y - v.y)^2 + (u.z - v.z)^2))
-    err <= max(atol, rtol*max(abs(u.t), abs(v.t), sqrt(u.x^2 + u.y^2 + u.z^2), sqrt(v.x^2 + v.y^2 + v.z^2)))
-end
+#export ΔR, Δϕ, Δη
+export fromPtEtaPhiE
 
-include("cartesian.jl")
-include("cylindrical.jl")
 
-# conversion
-function LorentzVector(v::LorentzVectorCyl)
-    x = px(v)
-    y = py(v)
-    z = pz(v)
-    t = energy(v)
-    return LorentzVector(t, x, y, z)
-end
+using LorentzVectorBase
+using InteractiveUtils
```

Comment:
what is this for?

Reply:
Right, this is necessary to get the supported coordinate systems. If there is a better way to get them, I am open to remove the dependency. Maybe, this should be moved to `LorentzVectorBase.jl` 🤔

Reply:
that seems quite brittle right? because it depends on what the it saw during pre-compile, let's not do that